From 856819b211a68c28857b215fe1e9069466eacc25 Mon Sep 17 00:00:00 2001
From: Allen Ye <allen.ye@mediatek.com>
Date: Thu, 11 Sep 2025 14:49:10 +0800
Subject: [PATCH 214/254] mtk: hostapd: FT: Add first_sta_seen flag for roamed
 sta

Add first_sta_seen flag prevent the group keys from being reset.

Signed-off-by: Allen Ye <allen.ye@mediatek.com>
---
 src/ap/ieee802_11.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 6a3135805..ace2360e7 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -4661,6 +4661,9 @@ static int __check_assoc_ies(struct hostapd_data *hapd, struct sta_info *sta,
 						       ies_len);
 			if (resp != WLAN_STATUS_SUCCESS)
 				goto out;
+			/* prevent the group keys from being reset after other sta
+			 * connect */
+			hapd->wpa_auth->group->first_sta_seen = true;
 		}
 #endif /* CONFIG_IEEE80211R_AP */
 
-- 
2.45.2

